---
title: "Leveraging Modern Data Stack in Box for Natural Product Genome Mining in Small-Scale and Private Strain Collection"
collection: talks
type: "Conference poster presentation"
permalink: /talks/2024-06-10-poster-1
venue: "Novo Nordisk Foundation Science Cluster"
date: 2024-06-10
location: "Copenhagen, Denmark"
---

![](
)This talk is about giving solutions to small and private labs to help them perform, manage, and explore genomics information from their strain collection to look for interesting BGCS. This talk offers more of the data management vibe from our [published workflow](https://doi.org/10.1093/nar/gkae314).

| <img src="https://sciencecluster.dk/wp-content/uploads/datalifecycle_illustration-web-final-1.jpg" width="100%" height="100%">| ![](https://media.licdn.com/dms/image/D4D22AQHo3UAfEl7I3w/feedshare-shrink_1280/0/1718264689827?e=1721865600&v=beta&t=h9oa9UwmkP-6W0OVDo-8189PppUYYC1QP-Qttl6-nEs) | 
|:--:|:--:|
|[Conference details here](https://sciencecluster.dk/event/cross-cluster-collaboration-data-science-for-planetary-and-human-health/) | _Group picture with Pep and Anna from NPGM group_ | 

I presented this poster at the "Data Science for Planetary and Human Health: The True Life Cycle of Multi-Omics Data" conference from the Novo Nordisk Foundation Science Cluster. 

A heartfelt thank you to the committee for awarding me the poster prize for our work! ðŸŽ‰



## Abstract
>The advent of third-generation sequencing technologies, especially Oxford Nanopore Technology, enables individual researchers and small laboratories to affordably create and manage private microbial strain collections. This shift promises to accelerate natural product discovery by facilitating the mining of biosynthetic gene clusters (BGCs) from genomic sequences, an important step in unlocking novel pharmaceuticals, agrochemicals, and other industrially relevant compounds. As researchers embark on building and analyzing their own private collections, the challenge extends beyond managing large-scale public genomic datasets but also in providing solutions that cater to the analysis of smaller, more focused collections. Here, we present BGCFlow, a comprehensive genome mining workflow for the analysis of bacterial pangenomes. BGCFlow integrates a "modern data stack in a box," leveraging tools such as dbt, DuckDB, and Metabase to offer streamlined data engineering pipeline and efficient platform for the exploration and management of private strain collections. Each tool is selected for its unique capabilities: dbt for transforming data with simplicity and reproducibility, duckdb for its lightweight, in-process SQL database that facilitates fast analytical queries, and metabase for its user-friendly interface allowing both data scientists and lab researchers to visualize and interact with their data. By doing so, we aim to bridge the gap between the potential of genome mining and the practicalities of conducting such research at a scale that is both manageable and accessible to a broader scientific community.

## Poster

<iframe src="/files/NNF_Poster_Matin_Nuhamunada_07062024.pdf#toolbar=0" width="100%" height="600px">
</iframe>
